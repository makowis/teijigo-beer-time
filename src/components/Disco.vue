<template>
  <div class="disco">
    <h1>DISCOGRAPHY</h1>
    <p>★★★★★★★工事中★★★★★★★</p>
    <img src="../assets/job_kouji_ojigi.png" alt="工事作業員のイラスト「おじぎ・お詫び」">
  </div>
</template>

<script lang="ts">
import Vue from 'vue';

export default Vue.extend({
  name: 'disco',
});
</script>

<style scoped>
</style>
