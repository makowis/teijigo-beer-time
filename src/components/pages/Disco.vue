<template>
  <tbt-default>
    <div class="disco">
      <tbt-h1>DISCOGRAPHY</tbt-h1>
      <tbt-section>
        <tbt-h2 :class="$style.albumTitle">
          2nd Mini Album 「START」
          <tbt-new-blink />
        </tbt-h2>
        <p :class="$style.onSale">
          <tbt-blink>2019.04.27 ON SALE！</tbt-blink>
        </p>
        <tbt-start-jacket-img :class="$style.jacket" />
        <table border="1" :class="$style.table">
          <thead>
            <tr>
              <th>No</th>
              <th>曲名</th>
              <th>アーティスト</th>
              <th>クレジット</th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td :class="$style.trackNo">1</td>
              <td>桜風狂想</td>
              <td>Pecan nene.</td>
              <td>
                作曲：ビール飲まない社員K<br />
                作詞：ビール飲まない社員K、みーやん<br />
                歌：みーやん
              </td>
            </tr>
            <tr>
              <td :class="$style.trackNo">2</td>
              <td>未来ヒーロー</td>
              <td>マッキー</td>
              <td>
                作詞・作曲：マッキー<br />
                歌：IA
              </td>
            </tr>
            <tr>
              <td :class="$style.trackNo">3</td>
              <td>ストロボサイン</td>
              <td>tan3-16</td>
              <td>
                作詞・作曲：u13<br />
                編曲：comorebi notes<br />
                歌：フナモトアユミ
              </td>
            </tr>
            <tr>
              <td :class="$style.trackNo">4</td>
              <td>桜花</td>
              <td>珍珠奶茶</td>
              <td>
                企画：WestTail<br />
                作曲：comorebi notes<br />
                作詞：台湾豆乳を飲む人<br />
                歌：Shiho (the Clock Hands)
              </td>
            </tr>
          </tbody>
        </table>
        <tbt-start-cross-fade :class="$style.crossFade" />
      </tbt-section>
      <hr />
      <tbt-section>
        <tbt-h2 :class="$style.albumTitle">1st Mini Album「HOPE」</tbt-h2>
        <p :class="$style.onSale">
          <tbt-blink>2018.10.28 ON SALE！</tbt-blink>
        </p>
        <tbt-hope-jacket-img :class="$style.jacket" />
        <table border="1" :class="$style.table">
          <thead>
            <tr>
              <th>No</th>
              <th>曲名</th>
              <th>アーティスト</th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td :class="$style.trackNo">1</td>
              <td>Alive</td>
              <td>WestTail feat. ruha</td>
            </tr>
            <tr>
              <td :class="$style.trackNo">2</td>
              <td>cresc.</td>
              <td>きむらじお feat.フナモトアユミ</td>
            </tr>
            <tr>
              <td :class="$style.trackNo">3</td>
              <td>瞬間</td>
              <td>山口真央 feat. すぴか</td>
            </tr>
            <tr>
              <td :class="$style.trackNo">4</td>
              <td>Calling</td>
              <td>NY2 feat. フナモトアユミ</td>
            </tr>
            <tr>
              <td :class="$style.trackNo">5</td>
              <td>Diary</td>
              <td>u13 feat. フナモトアユミ</td>
            </tr>
            <tr>
              <td :class="$style.trackNo">6</td>
              <td>ハルカユーグレハナヤイデ</td>
              <td>comorebi notes</td>
            </tr>
          </tbody>
        </table>
        <tbt-hope-cross-fade :class="$style.crossFade" />
      </tbt-section>
    </div>
  </tbt-default>
</template>

<script lang="ts">
import Vue from 'vue';
import Component from 'vue-class-component';
import TbtDefault from '@/components/templates/TbtDefault.vue';
import TbtH1 from '@/components/atoms/TbtH1.vue';
import TbtH2 from '@/components/atoms/TbtH2.vue';
import TbtSection from '@/components/atoms/TbtSection.vue';
import TbtBlink from '@/components/atoms/TbtBlink.vue';
import TbtNewBlink from '@/components/atoms/TbtNewBlink.vue';
import TbtHopeJacketImg from '@/components/atoms/TbtHopeJacketImg.vue';
import TbtStartJacketImg from '@/components/atoms/TbtStartJacketImg.vue';
import TbtHopeCrossFade from '@/components/atoms/TbtHopeCrossFade.vue';
import TbtStartCrossFade from '@/components/atoms/TbtStartCrossFade.vue';

@Component({
  components: {
    TbtDefault,
    TbtH1,
    TbtH2,
    TbtSection,
    TbtHopeJacketImg,
    TbtStartJacketImg,
    TbtBlink,
    TbtNewBlink,
    TbtHopeCrossFade,
    TbtStartCrossFade,
  },
})
export default class Disco extends Vue {}
</script>

<style module>
.table {
  margin: auto;
  text-align: left;
}

.table th,
.table td {
  padding: 5px 10px;
}

.trackNo {
  text-align: right;
}

.jacket {
  width: 400px;
  max-width: 70%;
  margin-bottom: 20px;
}

.albumTitle {
  margin-bottom: 5px;
}

.onSale {
  color: red;
  font-weight: bold;
  margin: 0 0 15px;
}

.crossFade {
  max-width: 80%;
  margin: 30px auto 30px;
}
</style>
