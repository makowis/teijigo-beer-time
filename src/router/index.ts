import Vue from 'vue';
import Router from 'vue-router';
import Top from '@/components/pages/Top.vue';
import Disco from '@/components/pages/Disco.vue';
import Bbs from '@/components/pages/Bbs.vue';
import Member from '@/components/pages/Member.vue';
import Beenya from '@/components/pages/Beenya.vue';
import NotFound from '@/components/pages/NotFound.vue';

Vue.use(Router);

export default new Router({
  mode: 'history',
  routes: [
    {
      path: '/',
      name: 'Top',
      component: Top,
    },
    {
      path: '/disco',
      name: 'Disco',
      component: Disco,
    },
    {
      path: '/bbs',
      name: 'Bbs',
      component: Bbs,
    },
    {
      path: '/member',
      name: 'Member',
      component: Member,
    },
    {
      path: '/beenya',
      name: 'Beenya',
      component: Beenya,
    },
    {
      path: '*',
      name: 'NotFound',
      component: NotFound,
    },
  ],
  scrollBehavior(to, from, savedPosition) {
    if (savedPosition) {
      return savedPosition;
    }
    return { x: 0, y: 0 };
  },
});
