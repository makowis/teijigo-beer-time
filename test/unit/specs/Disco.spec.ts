import Vue from 'vue';
import Disco from '@/components/pages/Disco.vue';
import router from '@/router/index';

describe('Disco.vue', () => {
  const Constructor = Vue.extend(Disco);
  const vm = new Constructor({ router }).$mount();

  it('should match the snapshot', () => {
    expect(vm.$el).toMatchSnapshot();
  });
});
