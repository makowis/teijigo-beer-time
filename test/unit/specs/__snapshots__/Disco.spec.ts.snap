// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Disco.vue should match the snapshot 1`] = `
<div
  id="app"
>
  <nav
    class="menuContainer"
    id="menu-container"
  >
    <ul
      class="menu"
    >
      <li
        class="menuItem"
      >
        <a
          class="router-link-exact-active router-link-active"
          href="/"
        >
          
        トップ
      
        </a>
      </li>
       
      <li
        class="menuItem"
      >
        <a
          class=""
          href="/disco"
        >
          
        DISCOGRAPHY
      
        </a>
      </li>
       
      <li
        class="menuItem"
      >
        <a
          class=""
          href="/member"
        >
          
        メンバー紹介
      
        </a>
      </li>
       
      <li
        class="menuItem"
      >
        <a
          class=""
          href="/bbs"
        >
          
        一言掲示板
      
        </a>
      </li>
       
      <li
        class="menuItem"
      >
        <a
          class=""
          href="/beenya"
        >
          
        びーにゃの部屋
      
        </a>
      </li>
       
      <li
        class="menuItem"
      >
        <a
          href="https://teijigo.booth.pm/items/1093403"
          target="_blank"
        >
          
        通販(BOOTH)
      
        </a>
      </li>
    </ul>
  </nav>
   
  <div
    class="mainContainer"
    id="main-container"
  >
    <a
      class="router-link-exact-active router-link-active"
      href="/"
    >
      <img
        class="logo"
        src="@/assets/logo_1.png"
      />
    </a>
     
    <div
      class="disco"
    >
      <h1
        class="h1"
      >
        DISCOGRAPHY
      </h1>
       
      <section
        class="section"
      >
        <h2
          class="h2 albumTitle"
        >
          
        2nd Mini Album 「START」
        
          <span
            class="blink newsBlink"
          >
            NEW
          </span>
        </h2>
         
        <p
          class="onSale"
        >
          <span
            class="blink"
          >
            2019.04.27 ON SALE！
          </span>
        </p>
         
        <img
          alt="2st Mini Album「Start」のジャケット画像"
          class="jacket"
          src="@/assets/2st_mini_album_start_jacket.png"
        />
         
        <table
          border="1"
          class="table"
        >
          <thead>
            <tr>
              <th>
                No
              </th>
               
              <th>
                曲名
              </th>
               
              <th>
                アーティスト
              </th>
               
              <th>
                クレジット
              </th>
            </tr>
          </thead>
           
          <tbody>
            <tr>
              <td
                class="trackNo"
              >
                1
              </td>
               
              <td>
                桜風狂想
              </td>
               
              <td>
                Pecan nene.
              </td>
               
              <td>
                
              作曲：ビール飲まない社員K
                <br />
                
              作詞：ビール飲まない社員K、みーやん
                <br />
                
              歌：みーやん
            
              </td>
            </tr>
             
            <tr>
              <td
                class="trackNo"
              >
                2
              </td>
               
              <td>
                未来ヒーロー
              </td>
               
              <td>
                マッキー
              </td>
               
              <td>
                
              作詞・作曲：マッキー
                <br />
                
              歌：IA
            
              </td>
            </tr>
             
            <tr>
              <td
                class="trackNo"
              >
                3
              </td>
               
              <td>
                ストロボサイン
              </td>
               
              <td>
                tan3-16
              </td>
               
              <td>
                
              作詞・作曲：u13
                <br />
                
              編曲：comorebi notes
                <br />
                
              歌：フナモトアユミ
            
              </td>
            </tr>
             
            <tr>
              <td
                class="trackNo"
              >
                4
              </td>
               
              <td>
                桜花
              </td>
               
              <td>
                珍珠奶茶
              </td>
               
              <td>
                
              企画：WestTail
                <br />
                
              作曲：comorebi notes
                <br />
                
              作詞：台湾豆乳を飲む人
                <br />
                
              歌：Shiho (the Clock Hands)
            
              </td>
            </tr>
          </tbody>
        </table>
         
        <div
          class="video crossFade"
        >
          <iframe
            allow="autoplay; encrypted-media"
            allowfullscreen="allowfullscreen"
            frameborder="0"
            height="315"
            src="https://www.youtube.com/embed/IxTdbkoGM-Q"
            width="560"
          />
        </div>
      </section>
       
      <hr />
       
      <section
        class="section"
      >
        <h2
          class="h2 albumTitle"
        >
          1st Mini Album「HOPE」
        </h2>
         
        <p
          class="onSale"
        >
          <span
            class="blink"
          >
            2018.10.28 ON SALE！
          </span>
        </p>
         
        <img
          alt="1st CD「HOPE」のジャケット画像"
          class="jacket"
          src="@/assets/1st_hope_jacket.png"
        />
         
        <table
          border="1"
          class="table"
        >
          <thead>
            <tr>
              <th>
                No
              </th>
               
              <th>
                曲名
              </th>
               
              <th>
                アーティスト
              </th>
            </tr>
          </thead>
           
          <tbody>
            <tr>
              <td
                class="trackNo"
              >
                1
              </td>
               
              <td>
                Alive
              </td>
               
              <td>
                WestTail feat. ruha
              </td>
            </tr>
             
            <tr>
              <td
                class="trackNo"
              >
                2
              </td>
               
              <td>
                cresc.
              </td>
               
              <td>
                きむらじお feat.フナモトアユミ
              </td>
            </tr>
             
            <tr>
              <td
                class="trackNo"
              >
                3
              </td>
               
              <td>
                瞬間
              </td>
               
              <td>
                山口真央 feat. すぴか
              </td>
            </tr>
             
            <tr>
              <td
                class="trackNo"
              >
                4
              </td>
               
              <td>
                Calling
              </td>
               
              <td>
                NY2 feat. フナモトアユミ
              </td>
            </tr>
             
            <tr>
              <td
                class="trackNo"
              >
                5
              </td>
               
              <td>
                Diary
              </td>
               
              <td>
                u13 feat. フナモトアユミ
              </td>
            </tr>
             
            <tr>
              <td
                class="trackNo"
              >
                6
              </td>
               
              <td>
                ハルカユーグレハナヤイデ
              </td>
               
              <td>
                comorebi notes
              </td>
            </tr>
          </tbody>
        </table>
         
        <div
          class="video crossFade"
        >
          <iframe
            allow="autoplay; encrypted-media"
            allowfullscreen="allowfullscreen"
            frameborder="0"
            height="315"
            src="https://www.youtube.com/embed/KE3It-tas0Y?rel=0"
            width="560"
          />
        </div>
      </section>
    </div>
     
    <footer
      class="footer"
    >
      <p>
        ©定時後ビールタイム
      </p>
    </footer>
  </div>
</div>
`;
